pkg/sql/catalog/descs/leased_descriptors.go

```diff
@@ -78,6 +79,14 @@ func (ld *leasedDescriptors) getByName(
 		return cached.(lease.LeasedDescriptor).Underlying(), false, nil
 	}
 
+	for _, d := range systemschema.UnleasableSystemDescriptors {
+		if parentID == d.GetParentID() &&
+			parentSchemaID == d.GetParentSchemaID() &&
+			name == d.GetName() {
+			return nil, true, nil
+		}
+	}
```

Comment:
I could replace this by looking up a `map[descpb.NameInfo]...` instead but since `systemschema.UnleasableSystemDescriptors` is always going to be tiny there really is no point and looping through the entries is perfectly fine.

Reply:
Anyway the point is, if the descriptor can't be leased, we just return here with the `shouldReadFromStore` return value set to `true`. This will cause the retrieval to fall back to KV.